docs/source/jit_language_reference_statements.rst

```diff
@@ -0,0 +1,389 @@
+                     ("," identifier ["as" identifier])* [","] ")"
+                     | "from" module "import" "*"
+    module          ::=  (identifier ".")* identifier
+    relative_module ::=  "."* module | "."+
+    `identifier_list <https://docs.python.org/3/reference/lexical_analysis.html#grammar-token-identifier>`
+
+The ``future`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    future_stmt ::=  "from" "__future__" "import" feature ["as" identifier]
+                 ("," feature ["as" identifier])*
+                 | "from" "__future__" "import" "(" feature ["as" identifier]
+                 ("," feature ["as" identifier])* [","] ")"
+    feature     ::=  identifier
+    `identifier <https://docs.python.org/3/reference/lexical_analysis.html#grammar-token-identifier>`
+
+
+Compound Statements
+~~~~~~~~~~~~~~~~~~~
+
+The following section describes the syntax of Compound statements that are supported in TorchScript.
+The section also highlights how Torchscript differs from regular Python statements.
+It is modeled after `the compound statements chapter of the Python language reference <https://docs.python.org/3/reference/compound_stmts.html>`_.
+
+The ``if`` statement:
+^^^^^^^^^^^^^^^^^^^^^
+* Torchscript supports both basic ``if/else`` and ternary ``if/else``.
+
+Basic ``if/else`` statement:
+""""""""""""""""""""""""""""
+
+::
+    if_stmt ::=  "if" assignment_expression ":" suite
+             ("elif" assignment_expression ":" suite)
+             ["else" ":" suite]
+    `assignment_expression <https://docs.python.org/3/reference/expressions.html#grammar-token-assignment-expression>`
+    `suite <https://docs.python.org/3/reference/compound_stmts.html#grammar-token-suite>`
+
+Ternary ``if/else`` statement:
+""""""""""""""""""""""""""""""
+
+::
+    if_stmt ::=  return [expression_list] "if" assignment_expression "else" [expression_list]
+    `assignment_expression <https://docs.python.org/3/reference/expressions.html#grammar-token-assignment-expression>`
+    `expression_list <https://docs.python.org/3/reference/expressions.html#grammar-token-expression-list>`
+
+* ``Tensor`` with 1 dimension is promoted to ``bool``
+    For eg.,
+    import torch
+
+    @torch.jit.script
+    def fn(x: torch.tensor):
+        if x: # The tensor gets promoted to bool
+            return True
+        return False
+
+    >> print(fn(torch.rand(1)))
+    >> True
+
+* ``Tensor`` with multi dimensions are not promoted to ``bool``
+    For eg.,:
+
+    import torch
+
+    # Multi dimensional Tensors error out.
+    # This below code gives RuntimeError:
+
+    @torch.jit.script
+    def fn():
+        if torch.rand(2):
+            print("Tensor is available")
+
+        if torch.rand(4,5,6):
+            print("Tensor is available")
+
+    >> print(fn())
+
+* If a condition variable is annotated as Final, either true or false branch is
+* evaluated depending on the  evaluation of the conditional variable.
```

Comment:
```suggestion
* If a conditional variable is annotated as Final, either true or false branch is
* evaluated depending on the evaluation of the conditional variable.
```

What is a conditional variable?

Reply:
An variable in a conditional expression like `if`/ `else`.